### ISSUE TITLE
how to set batchsize in valid in mode="train"

### ISSUE THREAD

**Issue by Taoboan1999** [ultralytics/ultralytics#20793]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

when i run YOLOv12, i set batch = 8. It is right when traing (2416 images ➗ 8=302)
100%|██████████| 302/302
but, when validation after one training epoch , i have 415 images. i see the output is
100%|██████████| 26/26
it must be batchsize=16. So how can i set a new valid batchsize

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        1/1      20.9G     0.8122     0.0449     0.6475        123        864: 100%|██████████| 302/302 [02:03<00:00, 
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 26/26 [00:1
                   all        415      31183      0.716      0.574      0.616      0.405

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Taoboan1999, thank you for reaching out and using Ultralytics 🚀! This is an automated response to help get you started. An Ultralytics engineer will also assist you soon.

We recommend visiting the [Docs](https://docs.ultralytics.com/) where you'll find plenty of [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many commonly asked questions may already be answered there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug your issue.

If you have a custom training ❓ Question (like batch size settings), please provide as much detail as possible—including dataset samples and logs—and review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join our community where it suits you best! For real-time chat, hop onto [Discord](https://discord.com/invite/ultralytics) 🎧. For deeper discussions, check out [Discourse](https://community.ultralytics.com/). Or join conversations on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Please upgrade to the latest `ultralytics` package (along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml)) in a [**Python>=3.8**](https://www.python.org/) environment and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue isn't already fixed:

```bash
pip install -U ultralytics
```

## Environments

YOLO runs in these up-to-date verified environments (with all dependencies, including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your patience and for being a part of the Ultralytics community! 💡

**Comment by Y-T-G** [comment]
It's not currently possible to set the validation batch size during training. It uses double the training batch size.

https://github.com/ultralytics/ultralytics/blob/7b07f4a1a192e4ed6b2d092519c8e74bf9c7eb42/ultralytics/engine/trainer.py#L314

**Comment by Taoboan1999** [comment]
> It's not currently possible to set the validation batch size during training. It uses double the training batch size.
> 
> [ultralytics/ultralytics/engine/trainer.py](https://github.com/ultralytics/ultralytics/blob/7b07f4a1a192e4ed6b2d092519c8e74bf9c7eb42/ultralytics/engine/trainer.py#L314)
> 
> Line 314 in [7b07f4a](/ultralytics/ultralytics/commit/7b07f4a1a192e4ed6b2d092519c8e74bf9c7eb42)
> 
>  batch_size=batch_size if self.args.task == "obb" else batch_size * 2,

Thank you very much for your patience you answer, if it is feasible I will try to modify this place to enable free control of the batchsize of val, thanks again!

**Comment by glenn-jocher** [comment]
Thanks for the clarification! You're welcome to modify that line to allow custom validation batch sizes. If you implement a solution, consider submitting a PR - it could be a useful feature for others as well.

For your modification, you might want to add a new argument like `val_batch` to the training configuration that defaults to the current behavior when not specified.
